spicy/toolchain/src/compiler/visitors/validator.cc

```diff
@@ -579,6 +580,22 @@ struct VisitorPost : public hilti::visitor::PreOrder<void, VisitorPost>, public
                 if ( a.tag() == "&size" || a.tag() == "&max-size" ) {
                     if ( ! a.hasValue() )
                         error(fmt("%s must provide an expression", a.tag()), p);
+                    else {
+                        auto v = hilti::visitor::PreOrder<>();
+                        for ( auto i : v.walk(a.value()) ) {
+                            auto&& member = i.node.tryAs<operator_::unit::MemberNonConst>();
+                            if ( ! member || ! member->hasOp0() )
+                                continue;
```

Comment:
Is this needed? Wouldn't just any usage of `self` enough for rejection? (i.e., just check if `i` is  the `ResolvedID`.)

Reply:
You are right, this e.g., fails to capture cases where `self` is passed to a function which accesses fields. I'll simplify this to reject all uses of `self`, regardless of context.